dtfs-central-api/api-tests/v1/utilisation-reports/delete-payment.api-test.ts

```diff
@@ -21,6 +25,9 @@ describe('DELETE /v1/utilisation-reports/:reportId/payment/:paymentId', () => {
   const tfmUser = aTfmUser();
   const tfmUserId = tfmUser._id.toString();
 
+  const facilityId = '11111111';
+  const facilityUtilisationData = FacilityUtilisationDataEntityMockBuilder.forId(facilityId).build();
```

Comment:
The fee record entity mock builder does create one, but this is just so that the lookup works correctly. We don't want cascade inserts to be enabled for fee records on facility utilisation data as we want to ensure fee records with invalid facility ids don't get uploaded, and part of the safety we get around that comes from the foreign key constraint. However, I do think it will get annoying very quickly that we need to do this, so I am thinking a good compromise here may be to update the `sql-db-helper` so that it handles inserting facility utilisation data where it doesn't exist for the fee records it needs to insert. What do you think?

Reply:
Ah I see... I guess I'm not completely convinced by the need to not have cascade inserts given we are validating existence. And anyway surely if we try and insert a fee record without a facility utilisation and an invalid facility id then the insert will fail and it has nothing to do with a cascade insert?

Also, how are we getting the facility ids in there for new facilities going forwards? I am guessing that insertion just going to happen completely separately from utilisation report uploads?

Reply:
I do accept turning on cascade inserts for test data is possibly a bad idea though. As for saving in sql-db-helper, I'm torn, I think if we do we should be explicit in the method name and have one that does it by default and one that doesn't. On the other hand for most PRs it will hopefully be a small thing to add, like the overhead for creating a report so if we don't want cascade inserts then I think this is not so bad.

Reply:
Discussed on call and agreed it should be safe enough to have cascade inserts and actually depending on future behaviour implementation decision there is a chance it could be necessary in mainline code anyway